### PR TITLE
On Windows run prospero using temporary jboss-modules to avoid resource locking

### PR DESCRIPTION
**Pull request by spyrkob**

### CODE REVIEW

**Comment by spyrkob** [comment]
@yersan let me know what you think, should the temporary jboss-modules be rather copied to `standalone|domain/tmp`?

**Comment by yersan** [comment]
@spyrkob I think we cross messages

> standalone|domain/tmp

I suggested just JBOSS_HOME, otherwise, we need to detect somehow if the user is running on standalone / domain and what is the temp server configured by the user.

File: dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat [hunk 1]
```diff
@@ -198,6 +198,18 @@ if exist "%PROSPERO_HOME%\jboss-modules.jar" (
   goto END
 )
 
+if "x%TMP%" == "x" (
+    echo Could not locate temporary folder using TMP env variable "%TMP%"
+    goto END
+) else (
+    goto COPY_JBOSS_MODULES
+)
```
[the remark this answers is not in the document]
Comment (spyrkob) [reply dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat:206]: you mean just create the temp file in `EAP_HOME`? Sure can do that, it gets cleaned up afterwards either way

[the remark this answers is not in the document]
Comment (yersan) [reply dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat:206]: yes, we can ensure there will be write permissions, when the server is running as a service generally run with a different user and I guess, that user could not have access to %TEMP%

File: dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat [hunk 2]
```diff
@@ -198,6 +198,13 @@ if exist "%PROSPERO_HOME%\jboss-modules.jar" (
   goto END
 )
 
+rem Use a copy of jboss-modules to avoid locking issues when jboss-modules is updated
+:COPY_JBOSS_MODULES
+    set "TMP_JBOSS_MODULES=%PROSPERO_HOME%\jboss-modules~%RANDOM%.tmp"
+    rem loop while we find a non-existing filename
+    if exist "%TMP_JBOSS_MODULES%" goto :COPY_JBOSS_MODULES
+    cp "%RUNJAR%" "%TMP_JBOSS_MODULES%"
```
Comment (yersan) [inline dist/common/src/main/resources/packages/org.jboss.prospero/content/bin/prospero.bat:206]: @spyrkob I found this, `cp` is not a valid command on Windows, this should be `copy` instead

**Pull request merged**